File: core/src/main/java/com/example/core/CoreComponent.java
package com.example.core;

import com.example.net.NetModule;
import com.example.net.NetworkApi;
import dagger.Component;

import javax.inject.Singleton;

@Singleton
@Component(modules = {CoreModule.class, NetModule.class})
public interface CoreComponent {

    MyCoreThing coreThing();
    NetworkApi networkApi();

}

File: core/src/main/java/com/example/core/CoreComponent.kt
package com.example.core

import com.example.net.NetModule
import com.example.net.NetworkApi
import dagger.Component

import javax.inject.Singleton

@Singleton
@Component(modules = [CoreModule::class, NetModule::class])
interface CoreComponent {

    fun coreThing(): MyCoreThing
    fun networkApi(): NetworkApi

}
